src/Planner/PlannerJoinTree.cpp

```diff
@@ -1242,6 +1245,7 @@ void joinCastPlanColumnsToNullable(QueryPlan & plan_to_add_cast, PlannerContextP
 }
 
 JoinTreeQueryPlan buildQueryPlanForJoinNode(const QueryTreeNodePtr & join_table_expression,
+    const QueryTreeNodePtr & nearest_query_node,
```

Comment:
Do you actually need this parameter?

Reply:
Indeed, initially I wanted not only to move expression from `ON` to `WHERE` here but in the opposite direction as well, so I needed to have corresponding query scope to access to `WHERE` query tree. However, let's avoid implementing in in that PR, maybe revisit after JoinStep refactoring, so here only focus on removal of `allow_experimental_join_condition`, so I've reverted that part of changes